### PR TITLE
Use stackalloc in System.Xml.Xsl.Runtime.NumberFormatterBase.ConvertToAlphabetic

### PR DESCRIPTION
**Pull request by TrayanZapryanov**
Another usage of new char[7] found and replaced with stackalloc

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Runtime/NumberFormatter.cs [hunk 1]
```diff
@@ -82,7 +82,7 @@ public static void ConvertToAlphabetic(StringBuilder sb, double val, char firstC
             Debug.Assert(1 <= val && val <= MaxAlphabeticValue);
             Debug.Assert(Math.Pow(totalChars, MaxAlphabeticLength) >= MaxAlphabeticValue);
 
-            char[] letters = new char[MaxAlphabeticLength];
+            Span<char> letters = stackalloc char[MaxAlphabeticLength];
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/Runtime/NumberFormatter.cs:85]: Unrelated to your change, but it'd be nice to use Math.DivRem below.

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Runtime/NumberFormatter.cs [hunk 2]
```diff
@@ -93,7 +93,7 @@ public static void ConvertToAlphabetic(StringBuilder sb, double val, char firstC
                 number = quot;
             }
             letters[--idx] = (char)(firstChar + --number);
-            sb.Append(letters, idx, MaxAlphabeticLength - idx);
+            sb.Append(letters.Slice(idx, MaxAlphabeticLength - idx));
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/Runtime/NumberFormatter.cs:96]: Is the `, MaxAlphabeticLength - idx` part needed? It couldn't just be `sb.Append(letters.Slice(id))`?

Comment (TrayanZapryanov) [reply src/libraries/System.Private.Xml/src/System/Xml/Xsl/Runtime/NumberFormatter.cs:96]: I think it is the same , just wanted to keep the code almost the same as before.

**Review by krwq — APPROVED** [review]
LGTM I'd recommend to apply changes suggested by @stephentoub, please let me know if you plan to leave as is

**Comment by krwq** [comment]
@TrayanZapryanov I'll merge as is in that case. If you feel like making the change feel free to create follow up PR. For the tests I'm currently not sure - I'd probably put a breakpoint in that code path and start running all tests and see the stacktrace. Optionally just always throw an exception from that method and see which tests fail.

**Pull request merged**